internal/scorecard/alpha/tests/bundle_test.go

```diff
@@ -143,3 +145,67 @@ func TestCRDHaveValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestOwnedCRDsForResources(t *testing.T) {
+	crdWithResources := operators.CustomResourceDefinitions{
+		Owned: []operators.CRDDescription{
+			operators.CRDDescription{
+				Name:              "Test",
+				Version:           "v1",
+				Kind:              "Test",
+				StatusDescriptors: make([]operators.StatusDescriptor, 0),
+				Resources: []operators.APIResourceReference{
+					operators.APIResourceReference{
+						Name:    "operator",
+						Kind:    "Test",
+						Version: "v1",
+					},
+				},
+			},
+		},
+	}
+
+	crdWithoutResources := operators.CustomResourceDefinitions{
+		Owned: []operators.CRDDescription{
+			operators.CRDDescription{
+				Name:              "Test",
+				Version:           "v1",
+				Kind:              "Test",
+				StatusDescriptors: make([]operators.StatusDescriptor, 0),
+				Resources:         make([]operators.APIResourceReference, 0),
+			},
+		},
+	}
+
+	tests := []struct {
+		name string
+		args operators.CustomResourceDefinitions
+		want scapiv1alpha2.State
+	}{
+		{
+			name: "Owned CRD with resources",
+			args: crdWithResources,
+			want: scapiv1alpha2.PassState,
+		},
+		{
+			name: "Owned CRD without resources",
+			args: crdWithoutResources,
+			want: scapiv1alpha2.FailState,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res := scapiv1alpha2.ScorecardTestResult{
+				Name:        "Check if CRDs have resources",
+				State:       scapiv1alpha2.PassState,
+				Errors:      make([]string, 0),
+				Suggestions: make([]string, 0),
+			}
```

Comment:
Why it is required?

Reply:
In the case of `CheckResources` we are passing only the crd and the `ScorecardTestResult`. The instantiation of `ScorecardResultSet` happens here: CRDsHaveResourcesTest. If we pass the resultSet without instantiating the values in `CheckResources`, it would be empty after the tests. Hence have instantiated it.